substra/sdk/backends/remote/backend.py

```diff
@@ -109,26 +113,103 @@ def add(self, spec, exist_ok=False, spec_options=None):
                 spec_options=spec_options,
             )
+        elif asset_type == schemas.Type.ComputePlan and spec_options.pop(AUTO_BATCHING, False):
+            if not batch_size:
+                raise ValueError("Batch size must be defined to create a compute plan \
+                    with the auto-batching feature.")
+            # Compute plan auto batching feature
+            return self._auto_batching_compute_plan(
+                spec=spec,
+                exist_ok=exist_ok,
+                batch_size=batch_size,
+                spec_options=spec_options,
+            )
 
         with spec.build_request_kwargs(**spec_options) as (data, files):
             response = self._add(asset_type, data, files=files, exist_ok=exist_ok)
         # The backend has inconsistent API responses when getting or adding an asset
         # (with much less data when responding to adds).
         # A second GET request hides the discrepancies.
+        # Do not do this with a compute plan or we lose the id_to_key field
         if asset_type == schemas.Type.ComputePlan:
             return response
 
         key = _get_asset_key(response)
         return self.get(asset_type, key)
 
-    def update_compute_plan(self, compute_plan_id, spec):
-        return self._client.request(
-            'post',
-            schemas.Type.ComputePlan.to_server(),
-            path=f"{compute_plan_id}/update_ledger/",
-            json=spec.dict(exclude_none=True),
+    def _auto_batching_compute_plan(self,
+                                    spec,
+                                    batch_size,
+                                    compute_plan_id=None,
+                                    exist_ok=None,
+                                    spec_options=None):
+        """Auto batching of the compute plan tuples
+        """
+        spec_options = spec_options or dict()
+        batches = compute_plan.auto_batching(
+            spec,
+            is_creation=compute_plan_id is None,
+            batch_size=batch_size,
         )
 
+        # Create / update by batch
+        id_to_keys = dict()
+        asset = None
+        for tmp_spec in batches:
+            if compute_plan_id:
+                asset = self._client.request(
+                    'post',
+                    schemas.Type.ComputePlan.to_server(),
+                    path=f"{compute_plan_id}/update_ledger/",
+                    json=tmp_spec.dict(exclude_none=True),
+                )
+                id_to_keys.update(asset['IDToKey'])
+            else:
+                with tmp_spec.build_request_kwargs(**spec_options) as (data, files):
+                    asset = self._add(
+                        schemas.Type.ComputePlan,
+                        data,
+                        files=files,
+                        exist_ok=exist_ok
+                    )
+                compute_plan_id = asset['computePlanID']
+            id_to_keys.update(asset['IDToKey'])
```

Comment:
I think it would be more readable to move out of the for loop the creation method:

```
first_batch, update_batches = ...
create_cp(first_batch)
for batch in update_batches:
    update_cp(batch)
```

Reply:
Updated it. I also changed the function so that instead of calling self._client.request themselves, they call the `add` and `update_compute_plan` methods without autobatching